lib/services/lambda/index.js

```diff
@@ -150,37 +155,70 @@ exports.deploy = function (ownServiceContext, ownPreDeployContext, dependenciesD
 }
 
 exports.consumeEvents = function (ownServiceContext, ownDeployContext, producerServiceContext, producerDeployContext) {
-    //TODO - DynamoDB streams will differ from this model
     return new Promise((resolve, reject) => {
         winston.info(`${SERVICE_NAME} - Consuming events from service '${producerServiceContext.serviceName}' for service '${ownServiceContext.serviceName}'`);
         let functionName = ownDeployContext.eventOutputs.lambdaName;
         let producerServiceType = producerServiceContext.serviceType;
         let principal;
-        let sourceArn;
-        if (producerServiceType === 'sns') {
-            principal = producerDeployContext.eventOutputs.principal;
-            sourceArn = producerDeployContext.eventOutputs.topicArn;
-        }
-        else if (producerServiceType === 'cloudwatchevent') {
+        if(producerServiceType === 'dynamodb') {
             principal = producerDeployContext.eventOutputs.principal;
-            sourceArn = producerDeployContext.eventOutputs.eventRuleArn;
-        }
-        else if (producerServiceType === 'alexaskillkit') {
-            principal = producerDeployContext.eventOutputs.principal;
-        }
-        else if (producerServiceType === 'iot') {
-            principal = producerDeployContext.eventOutputs.principal;
-            sourceArn = iotDeployersCommon.getTopicRuleArn(producerDeployContext.eventOutputs.topicRuleArnPrefix, ownServiceContext.serviceName);
-        }
-        else {
-            return reject(new Error(`${SERVICE_NAME} - Unsupported event producer type given: ${producerServiceType}`));
-        }
-
-        return lambdaCalls.addLambdaPermissionIfNotExists(functionName, principal, sourceArn)
-            .then(permissionStatement => {
+            let tableStreamArn = producerDeployContext.eventOutputs.tableStreamArn;
+            let tableName = producerDeployContext.eventOutputs.tableName;
+            let lambdaConsumers = producerDeployContext.eventOutputs.lambdaConsumers;
+            let lambdaConsumer;
+            let policyStatementsToConsume = JSON.parse(util.readFileSync(`${__dirname}/lambda-dynamodb-stream-role-statements.json`));
+            policyStatementsToConsume[0].Resource = [];
+            policyStatementsToConsume[0].Resource.push(`arn:aws:dynamodb:${accountConfig.region}:${accountConfig.account_id}:table/${tableName}/stream/*`)
```

Comment:
The problem is that this has the * after it, where as the tableStreamArn has a datestamp after /stream/